### PR TITLE
Refactor date-picker button tests to use @testing-library/react

### PR DESCRIPTION
**Pull request by vikneshwar**
#### Changes proposed in this Pull Request

This PR refactors the `date-button` component to use `@testing-library/react` instead of `enzyme`.

Part of #63409.

#### Testing instructions

Verify tests still pass: `yarn run test-client client/my-sites/backup/backup-date-picker/test/date-button.jsx`

### CODE REVIEW

**Comment by elliottprogrammer** [comment]
I think the testing instructions are incorrect...
The testing instructions say to run: `yarn run test-client client/my-sites/domains/components/form/test`.
However to run the tests for this PR, we should be running: ` yarn run test-client client/my-sites/backup/backup-date-picker/test`, right?

**Review by elliottprogrammer — APPROVED** [review]
I'm not too familiar with @testing-library/react, but the tests are passing and the code looks reasonable and appears all good to me.
LGTM! 👍

**Comment by vikneshwar** [comment]
@elliottprogrammer Oops! Sorry for the mess up there, I copied the instructions from another similar PR but missed updating the path; thanks for pointing that out. I have updated the instructions now

File: client/my-sites/backup/backup-date-picker/test/date-button.jsx [hunk 1]
```diff
@@ -14,35 +17,21 @@ jest.mock( '../hooks', () => ( {
...
-
-			// mock the useDispatch hook that will pick up the tracks event
-			useDispatch.mockImplementation( () => checkTracksEvent );
-
-			// shallow render
-			const DatePicker = shallow( <DateButton selectedDate={ moment() } /> );
-			// simulate a click on the button
-			DatePicker.find( '.backup-date-picker__date-button-button' ).simulate( 'click' );
-		} ) );
+		fireEvent.click( screen.getByText( 'Select Date' ) );
```
Comment (flootr) [inline client/my-sites/backup/backup-date-picker/test/date-button.jsx]: Using [`@testing-library/user-event`](https://testing-library.com/docs/react-testing-library/migrate-from-enzyme/#simulate-user-events) would be preferred over `fireEvent`. In this case you'd use it like

```js
import userEvent from '@testing-library/user-event';

// ... then inside your test function
await userEvent.click( screen.getByText( 'Select Date' );

// ... or using the `setup` explicitly
const user = userEvent.setup();
await user.click( screen.getByText( 'Select Date' );
```
To me the latter makes more sense if you have more than one interaction in your test fn. Using `user-event`  does a bunch more than firing only a single event and is closer to how an actual user would interact with a component.

**Review by tyxla — COMMENTED** [review]
Nothing to add to @flootr's review, just wanted to say thanks for helping with the migration 🙌

File: client/my-sites/backup/backup-date-picker/test/date-button.jsx (hunk 1 above)
Comment (vikneshwar) [reply client/my-sites/backup/backup-date-picker/test/date-button.jsx]: Thanks for the suggestion @flootr , I have updated to use `userEvent`

**Review by flootr — APPROVED** [review]
LGTM 🚢  Thanks for helping with the migration 👍🏻

**Review by tyxla — APPROVED** [review]
LGTM, thanks 🚢

**Pull request merged**